x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/JobNodeSelector.java

```diff
@@ -174,7 +169,7 @@ public JobNodeSelector(ClusterState clusterState,
                 continue;
             }
             // Assuming the node is eligible at all, check loading
-            allocateByMemory = currentLoad.isUseMemory();
+            boolean allocateByMemory = currentLoad.isUseMemory();
```

Comment:
```suggestion
            boolean canAllocateByMemory = currentLoad.isUseMemory();
```

I think this variable should be renamed now that it's being used for a slightly different purpose.